visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp

```diff
@@ -55,14 +55,22 @@ void MotionPlanningFrame::planButtonClicked()
 void MotionPlanningFrame::executeButtonClicked()
 {
   ui_->execute_button->setEnabled(false);
-  planning_display_->addBackgroundJob(boost::bind(&MotionPlanningFrame::computeExecuteButtonClicked, this), "execute");
+  // executing the motion in the (single) background thread, blocks visual updates of the scene robot
+  boost::thread(boost::bind(&MotionPlanningFrame::computeExecuteButtonClicked, this));
```

Comment:
Without separate threads for this, the background-job-thread was mainly blocked with waiting for the trajectory execution to finish. Everything else going to the background job simply needs to wait as well - which doesn't make sense.
To have a more uniform "look&feel", I augmented `addBackgroundJob()` with an extra option `ownThread` defaulting to false. For some reason the use of this code got lost when cleaning up my messy local repo. However, you suggested to go with boost::thread only. I have a slight preference for that too. Please vote for an option!